### PR TITLE
Fix symlink bug detection

### PR DESCRIPTION
**Pull request by jonathanmetzman**

### CODE REVIEW

**Comment by oliverchang** [comment]
Can you add a bit more context for these changes in the PR description?

File: infra/experimental/SystemSan/SystemSan.cpp [hunk 1]
```diff
@@ -324,44 +330,56 @@ void inspect_for_arbitrary_file_open(pid_t pid, const user_regs_struct &regs) {
...
+                                 std::ios_base::binary);
     if (bombfile.fail())
       return "";
     std::stringstream stream;
     stream << bombfile.rdbuf();
     return stream.str();
 }
 
+// https://oss-fuzz.com/testcase-detail/4882113260552192
+void report_bug_in_process(std::string bug_type, pid_t pid) {
```
Comment (oliverchang) [inline infra/experimental/SystemSan/SystemSan.cpp:343]: Can you add some comments on why this is needed vs the original `report_bug` ?

File: infra/experimental/SystemSan/SystemSan.cpp [hunk 2]
```diff
@@ -324,44 +330,56 @@ void inspect_for_arbitrary_file_open(pid_t pid, const user_regs_struct &regs) {
...
-  std::cout << "COMMAND" << command << std::endl;
+
+  std::string command = "rm -f " + path + " && ln -s " + kEvilLinkBombfile + " " + path;
+  std::cout << "COMMAND " << command << std::endl;
   system(command.c_str());
 }
 
 void initialize_evil_link_bombfile() {
-  system(("printf 'original' > " + kEvilLinkBombfile).c_str());
+  std::string command = ("printf " + kEvilLinkBombfileContents + " > " +
```
Comment (oliverchang) [inline infra/experimental/SystemSan/SystemSan.cpp:378]: Can we add a TODO to write this in C++ rather then shelling out? 

Same for above in evil_openat_hook.

Comment (jonathanmetzman) [reply infra/experimental/SystemSan/SystemSan.cpp:378]: haha sure.

**Pull request merged**